jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/core/ls/JakartaDelegateCommandHandlerForJava.java

```diff
@@ -108,4 +129,135 @@ private static JakartaJavaCompletionParams createJakartaJavaCompletionParams(Lis
+		boolean commandConfigurationUpdateSupported = getBoolean(obj, "commandConfigurationUpdateSupported");
+		boolean resolveSupported = getBoolean(obj, "resolveSupported");
+		JakartaJavaCodeActionParams params = new JakartaJavaCodeActionParams();
+		params.setTextDocument(texdDocumentIdentifier);
+		params.setRange(range);
+		params.setContext(context);
+		params.setResourceOperationSupported(resourceOperationSupported);
+		params.setCommandConfigurationUpdateSupported(commandConfigurationUpdateSupported);
+		params.setResolveSupported(resolveSupported);
+		return params;
+	}
+
+	private static CodeAction resolveCodeActionForJava(List<Object> arguments, String commandId,
+			IProgressMonitor monitor) throws JavaModelException, CoreException {
+		// Create java code action parameter
+		CodeAction unresolved = createJakartaJavaCodeActionResolveParams(arguments, commandId);
+		// Return code action from the code action parameter
+		return PropertiesManagerForJava.getInstance().resolveCodeAction(unresolved, JDTUtilsLSImpl.getInstance(),
+				monitor);
+	}
+
+	private static CodeAction createJakartaJavaCodeActionResolveParams(List<Object> arguments, String commandId) {
+		Map<String, Object> obj = getFirst(arguments);
+		if (obj == null) {
+			throw new UnsupportedOperationException(String.format(
+					"Command '%s' must be called with one CodeAction argument!", commandId));
+		}
+		CodeAction codeAction = JSONUtility.toModel(obj, CodeAction.class);
+		if (codeAction == null) {
+			throw new UnsupportedOperationException(String.format(
+					"Command '%s' must be called with one CodeAction argument!", commandId));
+		}
+		CodeActionResolveData resolveData = JSONUtility.toModel(codeAction.getData(), CodeActionResolveData.class);
+		if (resolveData == null) {
+			throw new UnsupportedOperationException(String.format(
+					"Command '%s' must be called with a CodeAction that has CodeActionResolveData!", commandId));
+		}
+		codeAction.setData(resolveData);
+		return codeAction;
+	}
+
+	
+    /**
+     * Returns the publish diagnostics list for a given java file URIs.
+     *
+     * @param arguments JakartaDiagnosticsParams @see
+     *                  org.eclipse.lsp4jakarta.commons.JakartaDiagnosticsParams
+     * @param monitor
+     * @return list of diagnostics as
+     *         CompletableFuture<List<PublishDiagnosticsParams>>
+     */
+	private static List<PublishDiagnosticsParams> getDiagnosticsForJava(List<Object> arguments, String commandId,
+			IProgressMonitor monitor) throws JavaModelException {
+		// Create java diagnostics parameter
+		JakartaJavaDiagnosticsParams params = createJakartaJavaDiagnosticsParams(arguments, commandId);
+		// Return diagnostics from parameter
+		return PropertiesManagerForJava.getInstance().diagnostics(params, JDTUtilsLSImpl.getInstance(), monitor);
+    }
+    
+	/**
+	 * Returns the java diagnostics parameters from the given arguments map.
+	 *
+	 * @param arguments
```

Comment:
Since you added javadoc to this method, can you please add it also for the other methods and complete the param descriptions?